src/platform/qpg6100/qpg6100Config.cpp

```diff
@@ -268,7 +268,7 @@ CHIP_ERROR QPG6100Config::MapNVMError(qvStatus_t aStatus)
     case QV_STATUS_KEY_LEN_TOO_SMALL:
         return CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
     case QV_STATUS_INVALID_DATA:
-        return CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
+        return CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
```

Comment:
Is it OK for us to mix `CHIP_ERROR` and `CHIP_DEVICE_ERROR`?

Reply:
Seems to be a fairly standard approach in the platform specific Config code
https://github.com/project-chip/connectedhomeip/blob/master/src/platform/EFR32/EFR32Config.cpp
https://github.com/project-chip/connectedhomeip/blob/master/src/platform/K32W/K32WConfig.cpp
Seems like CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND is a special case

Reply:
The fact other code does this does not make it correct.

There are a couple problems with this.  First, nothing prevents collisions between the CHIP_ERROR and CHIP_DEVICE_ERROR space.  Mixing the two in a return seems inherently problematic.  Second, it seems reasonable for callers to be able to call `ErrorStr` for returned errors.  That's going to break here.

Should `include/platform/CHIPDeviceError.h` exist at all?  Why do we have a separate, orphaned error code space with only four types of device errors?

Reply:
Maybe @woody-apple or @sagar-apple know, I'm not familiar with the code

Reply:
I'd agree with removing that.

Reply:
OK, then I suggest:

* not using CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND here (stop perpetuating the problem)
* add an appropriate CHIP_ERROR for this case if none exists
* filing an issue to clean CHIP_DEVICE_ERROR elsewhere

Reply:
Without CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND our code is broken. The error is a special case used in the program flow to set the defaults

https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/internal/GenericConfigurationManagerImpl.cpp

Reply:
@msandstedt @mspang @bzbarsky-apple would it be okay just to find and replace the occurrences and move it forward that way?

Reply:
There is a fairly robust sideband conversation on the more general issue of the multiple error code spaces we have.  Given this, I'm fine with any solution here, including proceeding with this code as is.  Resolving this across the repo is certainly out of scope for this PR.